### PR TITLE
Address Rule failures reported by Nexus Staging

### PR DESCRIPTION
**Pull request by narape**
# TL;DR
Address Rule failures reported by Nexus Staging when trying to publish a release to maven central.

## Type
 - [X] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
When releasing to Maven central, the Nexus staging system reports the following failures 
```
[INFO] [ERROR] Rule failure while trying to close staging repository with ID "orgflyte-1004".
[INFO] [ERROR] 
[INFO] [ERROR] Nexus Staging Rules Failure Report
[INFO] [ERROR] ==================================
[INFO] [ERROR] 
[INFO] [ERROR] Repository "orgflyte-1004" failures
[INFO] [ERROR]   Rule "javadoc-staging" failures
[INFO] [ERROR]     * Missing: no javadoc jar found in folder '/org/flyte/***-examples-scala/0.3.8'
[INFO] [ERROR]     * Missing: no javadoc jar found in folder '/org/flyte/flyteidl-protos/0.3.8'
[INFO] [ERROR]   Rule "pom-staging" failures
[INFO] [ERROR]     * Invalid POM: /org/flyte/***-parent/0.3.8/***-parent-0.3.8.pom: Project name missing, Project description missing, Project URL missing
[INFO] [ERROR] 
[INFO] [ERROR] 
```
See the full errors [here](https://github.com/flyteorg/flytekit-java/runs/5582733025?check_suite_focus=true)

Nexus has [requirements](https://central.sonatype.org/publish/requirements/) imposed on projects and it is complaining that we don't fulfill those requirements. 

## Tracking Issue
https://github.com/flyteorg/flyte/issues/2086

## Follow-up issue
_NA_

### CODE REVIEW

**Comment by narape** [comment]
Superseded by https://github.com/flyteorg/flytekit-java/pull/78
